plugins/inlineDefs.js

```diff
@@ -0,0 +1,228 @@
+'use strict';
+
+var JSAPI = require('../lib/svgo/jsAPI.js');
+
+exports.type = 'full';
+exports.active = true;
+exports.description = 'inlines svg definitions';
+
+/**
+ * plugin options
+ * @typedef {{onlyUnique: boolean}} Params
+ */
+exports.params = {
+    onlyUnique: true
+};
+
+
+/**
+ * replaces use tag with the corresponding definitions
+ * if onlyUnique = true, replaces only use tags with definitions referred to only once
+ * @param {Object} document
+ * @param {Params} params
+ * @returns {Object}
+ */
+exports.fn = function(document, params) {
+
+    var defs = document.querySelector('defs');
+    var uses = document.querySelectorAll( 'use' );
+
+    if(!uses) {
+        return document;
+    }
+
+    var useCount = _countUses(uses);
+
+    for (var i = 0; i < uses.length; i++) {
+        var hrefItem = uses[i].attr('xlink:href');
+        if (!hrefItem) {
+            hrefItem = uses[i].attr('href');
+        }
+        var href = hrefItem.value;
+
+        if (params.onlyUnique === true && useCount[href] > 1) {
+            continue;
+        }
+
+        var x = uses[i].hasAttr('x') ? uses[i].attr('x').value : null;
+        var y = uses[i].hasAttr('y') ? uses[i].attr('y').value : null;
+
+        var attrValue = null;
+        if (x && y) {
+            attrValue = 'translate(' + x + ', ' + y + ')';
+        } else if (x) {
+            attrValue = 'translate(' + x + ')';
+        }
+
+        var def = _findById(defs, href.match(idRegex)[1]);
+        if (!def) {
+            continue;
+        }
+        if (params.onlyUnique === true && useCount[href] === 1) {
+            def = _replaceElement(def);
+        }
+
+        for (var key in uses[i].attrs) {
+            if (uses[i].attrs.hasOwnProperty(key) && key !== 'x' && key !== 'y' && key !== 'xlink:href' && key !== 'href') {
+                def.addAttr(uses[i].attrs[key]);
+            }
+        }
+
+        if (attrValue) {
+            var g = new JSAPI({
+                elem: 'g',
+                attrs: {
+                    transform: {
+                        name: 'transform',
+                        value: attrValue,
+                        prefix: null,
+                        local: 'transform'
+                    }
+                },
+                content: [def]
+            });
+            _replaceElement(uses[i], g);
+        }
+        else {
+            _replaceElement(uses[i], def);
```

Comment:
If the element beeing inlined is a `symbol`-element, this line doesn't work.

I have the following SVG:

```xml
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="32" height="32">
    <defs>
        <symbol id="b" viewBox="0 0 31.55 27.98">
            <path fill="#fff" fill-rule="evenodd"
                  d="M31.36 25.74A1.51 1.51 0 0 1 30.05 28H1.5a1.46 1.46 0 0 1-.74-.19 1.5 1.5 0 0 1-.56-2.07L14.47.76a1.5 1.5 0 0 1 2.61 0z"/>
            <path fill="#ff9d00" fill-rule="evenodd"
                  d="M30.92 26L16.65 1a1 1 0 0 0-1.37-.37 1.09 1.09 0 0 0-.37.37L.63 26A1 1 0 0 0 1 27.35a1 1 0 0 0 .5.13h28.55a1 1 0 0 0 1-1 1 1 0 0 0-.13-.48z"/>
            <path d="M30.92 26a1 1 0 0 1 .13.49 1 1 0 0 1-1 1H1.5a1 1 0 0 1-.5-.13A1 1 0 0 1 .63 26L14.91 1a1.09 1.09 0 0 1 .37-.37 1 1 0 0 1 1.37.37zm-.43.25l-14.28-25a.49.49 0 0 0-.68-.18.47.47 0 0 0-.19.18l-14.27 25a.49.49 0 0 0 .18.68.54.54 0 0 0 .25.07h28.55a.5.5 0 0 0 .5-.5.5.5 0 0 0-.06-.26z"
                  opacity=".5"/>
        </symbol>
        <symbol id="a" viewBox="0 0 24.78 15.48">
            <path d="M16.78 3a2 2 0 1 0 2 2 2 2 0 0 0-2-2m-4-3a1 1 0 1 0 1 1 1 1 0 0 0-1-1m9 14a3 3 0 1 0-3-3 3 3 0 0 0 3 3m-2.69 1.48H.5a.54.54 0 0 1-.25-.06.51.51 0 0 1-.18-.69L8.28.36z"/>
        </symbol>
        <symbol id="c" viewBox="0 0 24.78 15.48">
            <use width="24.78" height="15.48" xlink:href="#a"/>
        </symbol>
    </defs>
    <use width="31.55" height="27.98" transform="translate(.2 3.02)" xlink:href="#b"/>
    <use width="24.78" height="15.48" transform="translate(1.22 14.5)" xlink:href="#c"/>
</svg>
```

Which should turn into something like this:

```xml
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="32" height="32">
    <g transform="translate(.2 3.02)">
        <path fill="#fff" fill-rule="evenodd" d="M31.36 25.74A1.51 1.51 0 0 1 30.05 28H1.5a1.46 1.46 0 0 1-.74-.19 1.5 1.5 0 0 1-.56-2.07L14.47.76a1.5 1.5 0 0 1 2.61 0z"/>
        <path fill="#ff9d00" fill-rule="evenodd" d="M30.92 26L16.65 1a1 1 0 0 0-1.37-.37 1.09 1.09 0 0 0-.37.37L.63 26A1 1 0 0 0 1 27.35a1 1 0 0 0 .5.13h28.55a1 1 0 0 0 1-1 1 1 0 0 0-.13-.48z"/>
        <path d="M30.92 26a1 1 0 0 1 .13.49 1 1 0 0 1-1 1H1.5a1 1 0 0 1-.5-.13A1 1 0 0 1 .63 26L14.91 1a1.09 1.09 0 0 1 .37-.37 1 1 0 0 1 1.37.37zm-.43.25l-14.28-25a.49.49 0 0 0-.68-.18.47.47 0 0 0-.19.18l-14.27 25a.49.49 0 0 0 .18.68.54.54 0 0 0 .25.07h28.55a.5.5 0 0 0 .5-.5.5.5 0 0 0-.06-.26z" opacity=".5"/>
    </g>
    <g transform="translate(1.22 14.5)">
        <g>
            <path d="M16.78 3a2 2 0 1 0 2 2 2 2 0 0 0-2-2m-4-3a1 1 0 1 0 1 1 1 1 0 0 0-1-1m9 14a3 3 0 1 0-3-3 3 3 0 0 0 3 3m-2.69 1.48H.5a.54.54 0 0 1-.25-.06.51.51 0 0 1-.18-.69L8.28.36z"/>
        </g>
    </g>
</svg>
```

As a proof of concept i cloned your repo and added the following code, which worked:

```
            if(def.elem === 'symbol') {
                def.elem = 'g';
                def.local = 'g';
                def.removeAttr('viewBox');
                def.removeAttr('width');
                def.removeAttr('height');
                def.removeAttr('id');
            }
            _replaceElement(uses[i], def);
```

I have no idea if this is the way to do it, but atleast my SVG worked with this modification.